cpp/src/arrow/compute/kernels/aggregate_basic_internal.h

```diff
@@ -164,6 +195,20 @@ struct MeanImpl : public SumImpl<ArrowType, SimdLevel> {
   using SumImpl<ArrowType, SimdLevel>::options;
 };
 
+struct NullMeanImpl : public NullSumImpl {
+  explicit NullMeanImpl(const ScalarAggregateOptions& options_) : NullSumImpl(options_) {}
+
+  Status Finalize(KernelContext*, Datum* out) override {
+    if ((options.skip_nulls || this->is_empty) && options.min_count == 0) {
+      // Return 0 if the remaining data is empty
+      out->value = std::make_shared<DoubleScalar>(0);
+    } else {
+      out->value = MakeNullScalar(float64());
```

Comment:
`mean(null)` is different with `sum(null)` as the returned value should be a `DoubleScalar`